### PR TITLE
Add main.tf

### PR DESCRIPTION
**Pull request by gerryhzgtest2021**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    tags = {
+        Name = "main"
+    }
+}
+
+#Subnets
+resource "aws_subnet" "main-public" {
+    count = 2
+    vpc_id = "${aws_vpc.main.id}"
+    cidr_block = "10.0.${count.index+1}.0/24"
```
Comment (vladimirmukhin) [inline main.tf:18]: This is a nice approach, however, it's not quite scalable.
It works well only if we have 10.0.0.0 as an initial subnet range.
Instead, use the list for storing values that depend on count.
```
locals {
  public_cidr_blocks = ["10.0.0.0/24", "10.0.0.0/24"]
}

resource "aws_subnet" "main-public" {
    count = 2

    vpc_id = "${aws_vpc.main.id}"
    cidr_block = local.public_cidr_blocks[count.index]
 }

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    map_public_ip_on_launch = "true"
+    availability_zone = "${count.index==0 ? "us-east-1a" : "us-east-1b"}"
```
Comment (vladimirmukhin) [inline main.tf:20]: List should be used here as well

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    tags = {
+        Name = "main-public-${count.index+1}"
+    }
+}
+
+resource "aws_subnet" "main-private" {
+    count = 2
```
Comment (vladimirmukhin) [inline main.tf:27]: Please separate `count` from other arguments using a newline
https://www.terraform.io/docs/language/syntax/style.html

File: main.tf [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    tags = {
+        Name = "main-public"
+    }
+}
+
+#Public subnets route table association
+resource "aws_route_table_association" "main-public" {
+    count = 2
+    subnet_id = "${aws_subnet.main-public[count.index].id}"
+    route_table_id = "${aws_route_table.main-public.id}"
```
Comment (vladimirmukhin) [inline main.tf:61]: When multiple arguments with single-line values appear on consecutive lines at the same nesting level, align their equals signs.
This is considered a good practice.
https://www.terraform.io/docs/language/syntax/style.html

File: main.tf [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    availability_zone = "${count.index==0 ? "us-east-1a" : "us-east-1b"}"
+    tags = {
+        Name = "main-private-${count.index+1}"
+    }
+}
+
+#Internet GW
+resource "aws_internet_gateway" "main-gw" {
+    vpc_id = "${aws_vpc.main.id}"
+    tags = {
```
Comment (vladimirmukhin) [inline main.tf:40]: It's good to separate `tag` from the rest of the arguments using blank line as well

File: main.tf [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    map_public_ip_on_launch = "true"
+    availability_zone = "${count.index==0 ? "us-east-1a" : "us-east-1b"}"
+    tags = {
+        Name = "main-public-${count.index+1}"
+    }
+}
+
+resource "aws_subnet" "main-private" {
+    count = 2
+    vpc_id = "${aws_vpc.main.id}"
```
Comment (vladimirmukhin) [inline main.tf:28]: Interpolation isn't required here.
Instead, use `aws_vpc.main.id`

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]
VPC itself is correct. Changes I've requested are mostly related to styling.
You can run the`terraform fmt` command to automatically fix most of them.

For each comment you address, please make sure that similar issues do not repeat in other parts of code as well.

Whenever you fix something, upload comit with the fix to the `task1` branch and mention in a comment that fix is done

File: main.tf (hunk 1 above)
Comment (gerryhzgtest2021) [reply main.tf]: fixed

File: main.tf (hunk 2 above)
Comment (gerryhzgtest2021) [reply main.tf]: fixed

File: main.tf (hunk 3 above)
Comment (gerryhzgtest2021) [reply main.tf]: fixed

File: main.tf (hunk 4 above)
Comment (gerryhzgtest2021) [reply main.tf]: fixed

File: main.tf (hunk 5 above)
Comment (gerryhzgtest2021) [reply main.tf]: fixed

File: main.tf (hunk 6 above)
Comment (gerryhzgtest2021) [reply main.tf]: fixed

**Pull request merged**